frontend/src/app/shared/components/autocompleter/op-autocompleter/op-autocompleter.component.ts

```diff
@@ -522,14 +522,6 @@ export class OpAutocompleterComponent<T extends IAutocompleteItem = IAutocomplet
       );
   }
 
-  private getDebounceTimeout():number {
-    if (this.initialDebounce) {
-      this.initialDebounce = false;
-      return 0;
-    }
```

Comment:
Removing the logic that disables the debounce on first user interaction - I don't think this behaviour makes sense (do we really want to make API calls for single letter query strings - in fact it would make more sense to me to only make requests when a minimum number of characters have been entered by the user)

Reply:
I vaguely remember that I added that to stabilize end-to-end tests. It was done in this commit: https://github.com/opf/openproject/commit/9aa36e54dbc4706041ce95badc6f859dfd69d429

Sadly, I don't remember much of it and I did not add a lot of comments or information in the commit message.

Reply:
could it have been related to the way `send_keys` work? on cuprite I think this directly sends a JS event

Reply:
No, can't be. Cuprite was added on 21 Jun 2023 in 4ce2ebacd0b. The initial debounce was added one month earlier on 24 May 2023 in 9aa36e5.

Reply:
Found it!

The loading indicator is set only after the first debounce completes. And in tests, we enter some text and wait for this loading indicator to disappear.

As it does not appear immediately, we do not wait for it, and so the test is out of sync with what is happening on screen.

That's probably why so many tests are failing on this PR.

Reply:
Found the reason in an old thread with Benjamin:

![image](https://github.com/user-attachments/assets/5555091f-a097-43ee-823b-01386d6d01b6)
![image](https://github.com/user-attachments/assets/9bf429a7-33ef-4c6d-bc53-cc7b58713bfa)
![image](https://github.com/user-attachments/assets/0d1dcfe8-3889-475d-b117-5118a07aba23)

And that's what I did: I set it to 0ms, but only for the first character being entered, so that the debouncing behavior is preserved.